lib/interactive_viewer.ml

```diff
@@ -60,9 +60,38 @@ let navigate z_patches (dir : direction) : unit =
   | Prev -> Lwd.set z_patches (Zipper.prev z)
   | Next -> Lwd.set z_patches (Zipper.next z)
 
-(* let pure_str s = Lwd.pure (W.string s) *)
 let quit = Lwd.var false
 
+let calculate_add_remove_operation z_patches : ui Lwd.t =
```

Comment:
Sorry if I wasn't clear! The organization in functions is not yet the one I had in mind.

We need to separate two distinct things:
- The computation of the two integers corresponding to the addition and removal
- and the creation of a ui element.

Ideally, the code for both should be in sufficiently distinct place, for instance two different function (or one function in another, but with a naming that suggests what it does.

But here, the `calculate_add_remove_operation` does not seem to return a UI element!

Reply:
@panglesd, Thank you for the clarification! I understand now. I'll work on separating the concerns into distinct functions:
1. One function will handle the computation of the integers corresponding to the addition and removal.
2. Another function will be responsible for creating the UI element based on those computed values.
I'll ensure that the code is organized in a way that reflects these separate responsibilities. Thank you for your patience!